frontend/src/metabase/parameters/utils/parameter-values.unit.spec.js

```diff
@@ -363,6 +346,55 @@ describe("parameters/utils/parameter-values", () => {
         "parameter2 default value",
       );
     });
+
+    describe("for number filter type", () => {
+      const numberParameter = {
+        id: 111,
+        slug: "numberParameter",
+        type: "number/=",
+      };
+
+      const runGetParameterValueFromQueryParams = value =>
+        getParameterValueFromQueryParams(
+          numberParameter,
+          {
+            [numberParameter.slug]: value,
+          },
+          metadata,
+        );
+
+      it("should parse the parameter value as a float when it is a number parameter without fields", () => {
+        expect(runGetParameterValueFromQueryParams("123.456")).toEqual([
+          123.456,
+        ]);
+      });
+
+      describe("when parsing parameter value that is a comma-separated list of numbers", () => {
+        it("should return list when every item is a number", () => {
+          expect(runGetParameterValueFromQueryParams("1,,2,3,4")).toEqual([
+            "1,2,3,4",
+          ]);
+          expect(runGetParameterValueFromQueryParams("1, ,2,3,4")).toEqual([
+            "1,2,3,4",
+          ]);
+          expect(runGetParameterValueFromQueryParams(",1,2,3,")).toEqual([
+            "1,2,3",
+          ]);
+        });
+
+        it("should return undefined when list is not formatted properly", () => {
+          expect(runGetParameterValueFromQueryParams(",,,")).toEqual([
+            undefined,
+          ]);
+          expect(runGetParameterValueFromQueryParams(" ")).toEqual([undefined]);
+        });
+
+        it("should return first parseable float if value includes non-numeric characters", () => {
+          expect(runGetParameterValueFromQueryParams("1,a,3,")).toEqual([1]);
+          expect(runGetParameterValueFromQueryParams("1a,b,3,")).toEqual([1]);
+        });
```

Comment:
Note that this test might change if we ever address https://github.com/metabase/metabase/issues/15295